addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestThermostatHandler.java

```diff
@@ -144,6 +174,28 @@ private void addUpdateRequest(String field, Object value) {
         addUpdateRequest(NEST_THERMOSTAT_UPDATE_URL, field, value);
     }
 
+    private void addTemperatureUpdateRequest(String celsiusField, String fahrenheitField,
+            QuantityType<Temperature> quantity) {
+        Unit<Temperature> unit = getTemperatureUnit();
+        BigDecimal value = quantityToRoundedTemperature(quantity, unit);
+        addUpdateRequest(NEST_THERMOSTAT_UPDATE_URL, unit == CELSIUS ? celsiusField : fahrenheitField,
+                value.doubleValue());
+    }
+
+    private Unit<Temperature> getTemperatureUnit() {
+        return getLastUpdate() != null && getLastUpdate().getTemperatureUnit() != null
+                ? getLastUpdate().getTemperatureUnit()
+                : CELSIUS;
+    }
+
+    private BigDecimal quantityToRoundedTemperature(QuantityType<Temperature> quantity, Unit<Temperature> unit)
+            throws IllegalArgumentException {
+        BigDecimal value = quantity.toUnit(unit).toBigDecimal();
```

Comment:
That's right @anthonygillet it will not convert any values when both your Thermostat and openHAB are configured to use the same unit. 

With the upstream PR a new "Measurement System" configuration option is introduced in  "Regional Settings". You can choose between the "SI" and "US" options.

> Maybe I am missing something though - are there cases that would cause the binding to do internal conversions of the API values from C to F or vice versa?

Conversion will only occur when:
* the Thermostat is using Fahrenheit and openHAB Celsius (or vice versa)
* a Fahrenheit command is sent while the unit of the Thermostat is unknown. It should then use Celsius as fallback. We could also opt for throwing an exception in that scenario.

It will now also round temperature values in case the precision (depending on the unit) does not match that of the Nest API.

Reply:
Awesome, that's what I thought the answer might be but I didn't want to guess.  Thanks for clearing this up for me!